### PR TITLE
v2trust set and show

### PR DESCRIPTION
**Pull request by baude**
add podman image trust set and show

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/images/trust.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"github.com/containers/libpod/cmd/podman/registry"
+	"github.com/containers/libpod/cmd/podman/validate"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	trustDescription = `Manages which registries you trust as a source of container images based on its location.
```
Comment (vrothberg) [inline cmd/podman/images/trust.go:11]:
```suggestion
	trustDescription = `Manages which registries you trust as a source of container images based on their location.
```

File: cmd/podman/images/trust.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  The location is determined by the transport and the registry host of the image.  Using this container image docker://docker.io/library/busybox as an example, docker is the transport and docker.io is the registry host.`
```
Comment (vrothberg) [inline cmd/podman/images/trust.go:12]: Can we use quay.io instead?

File: cmd/podman/images/trust_set.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Absolute path to keys is added to policy.json. May
+used multiple times to define multiple public keys.
+File(s) must exist before using this command`)
+	setFlags.StringVarP(&setOptions.Type, "type", "t", "signedBy", "Trust type, accept values: signedBy(default), accept, reject")
+}
+
+func setTrust(cmd *cobra.Command, args []string) error {
+	var (
+		validTrustTypes []string = []string{"accept", "insecureAcceptAnything", "reject", "signedBy"}
+	)
```
Comment (vrothberg) [inline cmd/podman/images/trust_set.go:46]: Nit pick: using the ` left-hand := right-hand` notation will spare two lines.

File: cmd/podman/images/trust_set.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	if len(args) != 1 {
+		return errors.Errorf("default or a registry name must be specified")
+	}
```
Comment (vrothberg) [inline cmd/podman/images/trust_set.go:49]: Can we use `Args:  cobra.ExactArgs(1),` in the command decl instead?

File: cmd/podman/images/trust_show.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	report, err := registry.ImageEngine().ShowTrust(registry.Context(), args, showTrustOptions)
+	if err != nil {
+		return err
+	}
+	if showTrustOptions.Raw {
+		fmt.Println(report.Raw)
+		return nil
+	}
+	if showTrustOptions.JSON {
+		//fmt.Print(string(report.JSONOutput))
```
Comment (vrothberg) [inline cmd/podman/images/trust_show.go:55]:
```suggestion
```

File: pkg/domain/entities/images.go [hunk 6]
```diff
@@ -4,6 +4,8 @@ import (
 	"net/url"
 	"time"
 
+	"github.com/containers/libpod/pkg/trust"
+
```
Comment (vrothberg) [inline pkg/domain/entities/images.go:8]: Drop empty line.

File: pkg/domain/entities/images.go [hunk 7]
```diff
@@ -285,3 +287,25 @@ type ImageTreeOptions struct {
...
+// ShowTrustReport describes the results of show trust
+type ShowTrustReport struct {
+	Raw                     []byte
+	SystemRegistriesDirPath string
+	JSONOutput              []byte
+	Policies                []*trust.TrustPolicy
+}
+
+type SetTrustOptions struct {
```
Comment (vrothberg) [inline pkg/domain/entities/images.go:307]: Missing comment.

File: pkg/trust/config.go [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package trust
+
+type TrustPolicy struct {
```
Comment (vrothberg) [inline pkg/trust/config.go:3]: Missing comment.

File: pkg/domain/infra/abi/trust.go [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+	}
+	for _, transval := range policyContentStruct.Transports {
+		for repo, repoval := range transval {
+			tempTrustShowOutput := trust.TrustPolicy{
+				Name:     repo,
+				RepoName: repo,
+				Type:     repoval[0].Type,
+			}
+			// TODO - keyarr is not used and I don't know its intent; commenting out for now for someone to fix later
+			//keyarr := []string{}
```
Comment (vrothberg) [inline pkg/domain/infra/abi/trust.go:139]:
```suggestion
```

File: pkg/domain/infra/abi/trust.go [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+			uids := []string{}
+			for _, repoele := range repoval {
+				if len(repoele.KeyPath) > 0 {
+					//keyarr = append(keyarr, repoele.KeyPath)
```
Comment (vrothberg) [inline pkg/domain/infra/abi/trust.go:143]:
```suggestion
```

File: pkg/domain/infra/abi/trust.go [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+					uids = append(uids, trust.GetGPGIdFromKeyPath(repoele.KeyPath)...)
+				}
+				if len(repoele.KeyData) > 0 {
+					//keyarr = append(keyarr, string(repoele.KeyData))
```
Comment (vrothberg) [inline pkg/domain/infra/abi/trust.go:147]:
```suggestion
```

File: pkg/domain/infra/abi/trust.go [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/libpod/pkg/trust"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
```
Comment (vrothberg) [inline pkg/domain/infra/abi/trust.go:15]: My internal brain parses had a hard time eating the code.

```suggestion
// TODO - the trust code below has (too?) many branches and lacks comments.
// While comments are needed for making the code more approachable, a code
// simplification may be worth to make it easier to maintain in the future.
```

File: cmd/podman/images/trust.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/images/trust.go:12]: How about docker://quay.io/podman/stable ?

File: pkg/domain/infra/abi/trust.go (hunk 9 above)
Comment (baude) [reply pkg/domain/infra/abi/trust.go:139]: this reflects the comment from above which isnt from me, i did a straight port ... im inclined to leave it.

File: pkg/domain/infra/abi/trust.go (hunk 10 above)
Comment (baude) [reply pkg/domain/infra/abi/trust.go:143]: this reflects the comment from above which isnt from me, i did a straight port ... im inclined to leave it.

File: pkg/domain/infra/abi/trust.go (hunk 11 above)
Comment (baude) [reply pkg/domain/infra/abi/trust.go:147]: this reflects the comment from above which isnt from me, i did a straight port ... im inclined to leave it.

File: pkg/domain/infra/abi/trust.go (hunk 12 above)
Comment (baude) [reply pkg/domain/infra/abi/trust.go:15]: this reflects the comment from above which isnt from me, i did a straight port ... im inclined to leave it.

**Comment by baude** [comment]
fixed up

File: cmd/podman/images/trust.go [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"github.com/containers/libpod/cmd/podman/registry"
+	"github.com/containers/libpod/cmd/podman/validate"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	trustDescription = `Manages which registries you trust as a source of container images based on their location.
+  The location is determined by the transport and the registry host of the image.  Using this container image docker://quay.io/podman/stable as an example, docker is the transport and docker.io is the registry host.`
```
Comment (TomSweeneyRedHat) [inline cmd/podman/images/trust.go:12]: "docker.io is the" -> "quay.io is the"

Comment (TomSweeneyRedHat) [inline cmd/podman/images/trust.go:12]:
```suggestion
  The location is determined by the transport and the registry host of the image.  Using this container image docker://quay.io/podman/stable as an example, docker is the transport and quay.io is the registry host.`
```

**Pull request merged**